src/cc2olx/models.py

```diff
@@ -0,0 +1,625 @@
+                                "identifier": "x" * 34,
+                                "title": unit.get("title", "none"),
+                                "children": [unit],
+                            }
+                            for unit in subsection.get("children", [])
+                        ]
+                    else:
+                        units = [
+                            {
+                                "identifier": "x" * 34,
+                                "title": "none",
+                                "children": subsection.get("children", []),
+                            },
+                        ]
+                else:
+                    units = subsection.get("children", [])
+                pprint(1, "subsection", subsection, len(units))
+                normal_subsection = {
+                    "children": [],
+                    "identifier": subsection.get("identifier"),
+                    "identifierref": subsection.get("identifierref"),
+                    "title": subsection.get("title"),
+                }
+                for unit in units:
+                    if is_leaf(unit):
+                        # Structure is too shallow.
+                        # Found leaf at unit level.
+                        components = [
+                            unit,
+                        ]
+                    else:
+                        components = unit.get("children", [])
+                        components = self.flatten(components)
+                    pprint(2, "unit", unit, len(components))
+                    normal_unit = {
+                        "children": [],
+                        "identifier": unit.get("identifier"),
+                        "identifierref": unit.get("identifierref"),
+                        "title": unit.get("title"),
+                    }
+                    for component in components:
+                        pprint(3, "component", component)
+                        normal_unit["children"].append(component)
+                    normal_subsection["children"].append(normal_unit)
+                normal_section["children"].append(normal_subsection)
+            normal_course["children"].append(normal_section)
+        self.normalized = normal_course
+        return normal_course
+
+    def flatten(self, container):
+        if is_leaf(container):
+            return container
+        if isinstance(container, list):
+            children = container
+        else:
+            # Structure is too deep.
+            # Flatten into current unit?
+            # Found non-leaf at component level
+            children = container.get("children", [])
+        output = []
+        for child in children:
+            if is_leaf(child):
+                output.append(child)
+            else:
+                leaves = self.flatten(child)
+                output.extend(leaves)
+        return output
+
+    def get_resource_content(self, identifier):
+        """
```

Comment:
Could we in general add more Docstrings to the methods and files to make it easier to understand the purposes of the files and methods included? 

I can also agree to have this added in future PRs. I know this is PR is mostly a port of the existing code.

Reply:
>Could we in general add more Docstrings to the methods and files to make it easier to understand the purposes of the files and methods included?

That's reasonable :+1: 
I'll definitely add more documentation.

>I can also agree to have this added in future PRs.

It would be nice to do this in following feature PR's because:
- Some code here likely will be refactored.
- I'm in progress of mastering the `imscc` and `olx` specifications, so probably can write better docs soon.